src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -105,6 +152,50 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
       .map(_.value)
       .sum
 
+  /**
+    * Get average fee for the specified wait time interval
+    * @param expectedWaitTimeMinutes maximal amount of time for which transaction can be kept in mempool
+    * @param txSize size of transaction (in bytes)
+    *  @return recommended fee value for transaction to be proceeded in specified time
+    */
+  def getRecommendedFee(expectedWaitTimeMinutes : Int, txSize : Int) : Long = {
+    if (expectedWaitTimeMinutes < MemPoolStatisticsParams.nHistogramBins) {
+      // locate first non-empty histogram bin preceding or equal to the specified wait time
+      for (i <- expectedWaitTimeMinutes to 0 by -1) {
+        val h = stats.histogram(expectedWaitTimeMinutes)
+        if (h.nTxns != 0) {
+          return h.totalFee / h.nTxns * txSize / 1024
+        }
+      }
+    }
+    settings.nodeSettings.minimalFeeAmount
+  }
```

Comment:
Can be moved out to a histogram builder and then unit-tested independently of this mempool.